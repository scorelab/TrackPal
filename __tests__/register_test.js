import React from 'react';
import SignIn from '../app/screens/SignUpScreen/signUpScreen.js';

import renderer from 'react-test-renderer';

jest.useFakeTimers();

test('renders correctly', () => {
  const tree = renderer.create(<SignIn />).toJSON();
  expect(tree).toMatchSnapshot();
});