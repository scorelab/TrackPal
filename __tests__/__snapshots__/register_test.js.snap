// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/landing.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.7)",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "bottom": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 100,
                    "justifyContent": "center",
                    "marginBottom": 5,
                    "marginTop": 20,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    animating={true}
                    animationDuration={1600}
                    animationEasing={[Function]}
                    color="black"
                    interaction={true}
                    renderComponent={[Function]}
                    style={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                    waveFactor={0.54}
                    waveMode="fill"
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "black",
                            "borderRadius": 20,
                            "borderWidth": 0,
                            "height": 40,
                            "opacity": 1,
                            "transform": Array [
                              Object {
                                "scale": 0,
                              },
                            ],
                            "width": 40,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "black",
                            "borderRadius": 20,
                            "borderWidth": 0,
                            "height": 40,
                            "opacity": 1,
                            "transform": Array [
                              Object {
                                "scale": 0,
                              },
                            ],
                            "width": 40,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "black",
                            "borderRadius": 20,
                            "borderWidth": 0,
                            "height": 40,
                            "opacity": 1,
                            "transform": Array [
                              Object {
                                "scale": 0,
                              },
                            ],
                            "width": 40,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "black",
                            "borderRadius": 20,
                            "borderWidth": 0,
                            "height": 40,
                            "opacity": 1,
                            "transform": Array [
                              Object {
                                "scale": 0,
                              },
                            ],
                            "width": 40,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.1)",
                    "borderRadius": 50,
                    "marginBottom": 10,
                    "padding": 40,
                    "paddingTop": 50,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Name"
                  placeholderTextColor="rgba(0,0,0,0.5)"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 5,
                      "color": "black",
                      "height": 45,
                      "marginBottom": 15,
                      "paddingHorizontal": 10,
                      "width": 600,
                    }
                  }
                  underlineColorAndroid="transparent"
                />
                <TextInput
                  allowFontScaling={true}
                  keyboardType="email-address"
                  onChangeText={[Function]}
                  placeholder="Email"
                  placeholderTextColor="rgba(0,0,0,0.5)"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 5,
                      "color": "black",
                      "height": 45,
                      "marginBottom": 15,
                      "paddingHorizontal": 10,
                      "width": 600,
                    }
                  }
                  underlineColorAndroid="transparent"
                />
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Pasword"
                  placeholderTextColor="rgba(0,0,0,0.5)"
                  secureTextEntry={true}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 5,
                      "color": "black",
                      "height": 45,
                      "marginBottom": 15,
                      "paddingHorizontal": 10,
                      "width": 600,
                    }
                  }
                  underlineColorAndroid="transparent"
                />
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Confirm Pasword"
                  placeholderTextColor="rgba(0,0,0,0.5)"
                  secureTextEntry={true}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 5,
                      "color": "black",
                      "height": 45,
                      "marginBottom": 15,
                      "paddingHorizontal": 10,
                      "width": 600,
                    }
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3c5898",
                "borderRadius": 5,
                "borderWidth": 0,
                "height": 45,
                "justifyContent": "center",
                "marginBottom": 15,
                "opacity": 1,
                "width": 600,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                }
              }
            >
              Sign Up
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "marginBottom": 5,
                }
              }
            >
              or
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              disabled={false}
              onLongPress={[Function]}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#3b5998",
                  "borderRadius": 30,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 7,
                  "paddingBottom": 14,
                  "paddingTop": 14,
                  "shadowColor": "rgba(0,0,0, .4)",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 1,
                  "width": 200,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="#3b5998"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 24,
                      },
                      Object {},
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                      "marginLeft": 15,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Sign Up With Facebook
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": 25,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "marginBottom": 5,
                  }
                }
              >
                Already have an account?
                <Text
                  style={
                    Object {
                      "color": "#0066cc",
                    }
                  }
                >
                   Sign In
                </Text>
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
